src/LuaPiGui.cpp

```diff
@@ -0,0 +1,1105 @@
+	int num = luaL_checkinteger(l, 1);
+	ImGui::PopStyleColor(num);
+	return 0;
+}
+
+static int l_pigui_push_style_var(lua_State *l) {
+	int style = luaL_checkImGuiStyleVar(l, 1);
+	float val = luaL_checknumber(l, 2);
+	ImGui::PushStyleVar(style, val);
+	return 0;
+}
+
+static int l_pigui_pop_style_var(lua_State *l) {
+	int num = luaL_checkinteger(l, 1);
+	ImGui::PopStyleVar(num);
+	return 0;
+}
+
+static int l_pigui_add_line(lua_State *l) {
+	ImDrawList* draw_list = ImGui::GetWindowDrawList();
+	ImVec2 a = luaL_checkImVec2(l, 1);
+	ImVec2 b = luaL_checkImVec2(l, 2);
+	ImColor col = luaL_checkImColor(l, 3);
+	double thickness = luaL_checknumber(l, 4);
+	draw_list->AddLine(a, b, col, thickness);
+	return 0;
+}
+
+static int l_pigui_add_circle(lua_State *l) {
+	ImDrawList* draw_list = ImGui::GetWindowDrawList();
+	ImVec2 center = luaL_checkImVec2(l, 1);
+	int radius = luaL_checkinteger(l, 2);
+	ImColor color = luaL_checkImColor(l, 3);
+	int segments = luaL_checkinteger(l, 4);
+	double thickness = luaL_checknumber(l, 5);
+	draw_list->AddCircle(center, radius, color, segments, thickness);
+	return 0;
+}
+
+static int l_pigui_add_circle_filled(lua_State *l) {
+	ImDrawList* draw_list = ImGui::GetWindowDrawList();
+	ImVec2 center = luaL_checkImVec2(l, 1);
+	int radius = luaL_checkinteger(l, 2);
+	ImColor color = luaL_checkImColor(l, 3);
+	int segments = luaL_checkinteger(l, 4);
+	draw_list->AddCircleFilled(center, radius, color, segments);
+	return 0;
+}
+
+static int l_pigui_path_arc_to(lua_State *l) {
+	ImDrawList* draw_list = ImGui::GetWindowDrawList();
+	ImVec2 center = luaL_checkImVec2(l, 1);
+	double radius = luaL_checknumber(l, 2);
+	double amin = luaL_checknumber(l, 3);
+	double amax = luaL_checknumber(l, 4);
+	int segments = luaL_checkinteger(l, 5);
+	draw_list->PathArcTo(center, radius, amin, amax, segments);
+	return 0;
+}
+
+static int l_pigui_path_stroke(lua_State *l) {
+	ImDrawList* draw_list = ImGui::GetWindowDrawList();
+	ImColor color = luaL_checkImColor(l, 1);
+	bool closed = luaL_checkbool(l, 2);
+	double thickness = luaL_checknumber(l, 3);
+	draw_list->PathStroke(color, closed, thickness);
+	return 0;
+}
+
+static int l_pigui_selectable(lua_State *l) {
+	std::string label = luaL_checkstring(l, 1);
+	bool selected = luaL_checkbool(l, 2);
+	ImGuiSelectableFlags flags = luaL_checkImGuiSelectableFlags(l, 3);
+	// TODO: parameter size
+	bool res = ImGui::Selectable(label.c_str(), selected, flags);
+	lua_pushboolean(l, res);
+	return 1;
+}
+
+static int l_pigui_text(lua_State *l) {
+	std::string text = luaL_checkstring(l, 1);
+	ImGui::Text(text.c_str());
+	return 0;
+}
+
+static int l_pigui_button(lua_State *l) {
+	std::string text = luaL_checkstring(l, 1);
+	bool ret = ImGui::Button(text.c_str());
+	lua_pushboolean(l, ret);
+	return 1;
+}
+
+static int l_pigui_text_wrapped(lua_State *l) {
+	std::string text = luaL_checkstring(l, 1);
+	ImGui::TextWrapped(text.c_str());
+	return 0;
+}
+
+// static int l_pigui_get_velocity(lua_State *l) {
```

Comment:
It depends on changes in Lua*.cpp files that I did not want to introduce yet. I can remove it and later add it again, just thought I'd leave it in so you can see what is yet to come :)